src/components/JobsContext.jsx

```diff
@@ -36,6 +38,22 @@ const JobsProvider = ({ children, client, options }) => {
     return new CozyRealtime({ client })
   }, [client])
 
+  const setJobsWithName = async arr => {
+    const promises = arr.map(async jobInProgress => {
+      const slug = jobInProgress.konnector
+      const resp = await client.query(
+        Q(KONNECTOR_DOCTYPE).getById(`${KONNECTOR_DOCTYPE}/${slug}`)
+      )
+      const name = get(resp, 'data.attributes.name')
+      const newJobInProgress = { ...jobInProgress }
+      newJobInProgress.institutionLabel = name
+      return newJobInProgress
+    })
+
+    const newJobsInProgress = await Promise.all(promises)
+    setJobsInProgress(newJobsInProgress)
+  }
```

Comment:
I think this particular business logic could be extracted to another provider, using JobsProvider underneath. This way we could extract JobsProvider for consumption in other non-banking contexts and keep the banking business logic separated from the generic jobs state logic.

Reply:
Instead of doing several requests to fetch the individual jobs, it would be better

* either to use Q().getByIds (with an s) to fetch several documents in one request
* or use the relationship mechanism to have the jobs doctype be linked via a relationship to konnectors, and use include(['konnectors']) in the query definition for them to be automatically fetched by cozy-client